libs/ui-lib/lib/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx

```diff
@@ -37,57 +37,80 @@ const AdvancedNetworkFields = () => {
     const isPrimaryIPv6 = Address6.isValid(primaryMachineNetworkCidr);
 
     // Check Cluster Networks
-    if (values.clusterNetworks && values.clusterNetworks.length >= 2) {
-      const firstClusterCidr = values.clusterNetworks[0].cidr;
-      if (!firstClusterCidr) return;
+    if (clusterNetworksRef.current && clusterNetworksRef.current.length >= 2) {
+      const first = clusterNetworksRef.current[0];
+      const second = clusterNetworksRef.current[1];
+      const firstClusterCidr = first?.cidr;
+      const secondClusterCidr = second?.cidr;
+      if (!firstClusterCidr || !secondClusterCidr) return;
 
       const isFirstClusterIPv6 = Address6.isValid(firstClusterCidr);
-
-      // If primary machine is IPv6 but first cluster is IPv4, swap them
-      if (isPrimaryIPv6 && !isFirstClusterIPv6) {
-        const reordered = [values.clusterNetworks[1], values.clusterNetworks[0]];
+      const isSecondClusterIPv6 = Address6.isValid(secondClusterCidr);
+
+      // Only swap when families are opposite and mismatched with primary
+      if (isFirstClusterIPv6 === isSecondClusterIPv6) {
+        // Both have the same family; don't attempt to reorder to avoid oscillation
+      } else if (isPrimaryIPv6 && !isFirstClusterIPv6 && isSecondClusterIPv6) {
+        const reordered = [...(clusterNetworksRef.current || [])];
+        [reordered[0], reordered[1]] = [reordered[1], reordered[0]];
         setFieldValue('clusterNetworks', reordered, false);
-      }
-      // If primary machine is IPv4 but first cluster is IPv6, swap them
-      else if (!isPrimaryIPv6 && isFirstClusterIPv6) {
-        const reordered = [values.clusterNetworks[1], values.clusterNetworks[0]];
+      } else if (!isPrimaryIPv6 && isFirstClusterIPv6 && !isSecondClusterIPv6) {
+        const reordered = [...(clusterNetworksRef.current || [])];
+        [reordered[0], reordered[1]] = [reordered[1], reordered[0]];
         setFieldValue('clusterNetworks', reordered, false);
       }
-    }
-
-    // Check Service Networks
-    if (values.serviceNetworks && values.serviceNetworks.length >= 2) {
-      const firstServiceCidr = values.serviceNetworks[0].cidr;
-      if (!firstServiceCidr) return;
 
-      const isFirstServiceIPv6 = Address6.isValid(firstServiceCidr);
-
-      // If primary machine is IPv6 but first service is IPv4, swap them
-      if (isPrimaryIPv6 && !isFirstServiceIPv6) {
-        const reordered = [values.serviceNetworks[1], values.serviceNetworks[0]];
-        setFieldValue('serviceNetworks', reordered, false);
-      }
-      // If primary machine is IPv4 but first service is IPv6, swap them
-      else if (!isPrimaryIPv6 && isFirstServiceIPv6) {
-        const reordered = [values.serviceNetworks[1], values.serviceNetworks[0]];
-        setFieldValue('serviceNetworks', reordered, false);
+      // Check Service Networks
+      if (serviceNetworksRef.current && serviceNetworksRef.current.length >= 2) {
```

Comment:
the swapping logic for both service and cluster networks is the same. You should make a generic function for network swapping (regardles if its cluster or service) which you would reuse.

Something like
```ts
const swapNetworks = (network: ServiceNetwork[] | ClusterNetwork[], fieldName: 'clusterNetworks' | 'serviceNetworks') => {
 ...
}
```